### PR TITLE
test(provider/kubernetes): v2 artifact tests

### PR DESCRIPTION
**Pull request by lwander**
This is just the start, will test a few more flows in subsequent PRs

### CODE REVIEW

File: testing/citest/tests/kube_smoke_test.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
 
 # Standard python modules.
 import sys
+import random
```
Comment (jtk54) [inline testing/citest/tests/kube_smoke_test.py]: Unused?

File: testing/citest/tests/kube_v2_artifact_test.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    """
+    super(KubeV2ArtifactTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']
+
+    # Take just the first if there are multiple
```
Comment (jtk54) [inline testing/citest/tests/kube_v2_artifact_test.py]: Does it matter which namespace this deploys to? Are there problems when multiple tests are targeting the same namespace?

File: testing/citest/tests/kube_smoke_test.py (hunk 1 above)
Comment (lwander) [reply testing/citest/tests/kube_smoke_test.py]: Whoops, yes

File: testing/citest/tests/kube_v2_artifact_test.py (hunk 2 above)
Comment (lwander) [reply testing/citest/tests/kube_v2_artifact_test.py]: Doesn't matter, we make sure all the names are unique enough

**Pull request merged**